shell/common/engine.h

```diff
@@ -767,6 +777,27 @@ class Engine final : public RuntimeDelegate,
   ///
   const std::string& InitialRoute() const { return initial_route_; }
 
+  //--------------------------------------------------------------------------
+  /// @brief      Loads the dart shared library into the dart VM. When the
```

Comment:
I guess it depends on what you would consider "loading". If you consider dlopen-ing a file and resolving symbols loading, then indeed it just passes the results of that on. On the other hand, this method does directly call Dart_DeferredLoadComplete, which "loads" the symbols into the running isolate. In this case, this method does invoke loading with the provided symbols.

Reply:
ya, I just realized my understanding was incorrect. I think in your diagram, there's actually another bubble somewhere in the Runtime/VM box that's for "consuming the bytes than represent the AOT program" like how you have a small bubble for dlopen etc. 

In that case, maybe this function should just be called loadDartDeferredLibrary too since the implementation behind this function (the vm) loads the program given to it.